libraries/scripting/dependencies/src/kotlin/script/experimental/dependencies/annotations.kt

```diff
@@ -33,22 +31,42 @@ annotation class Repository(vararg val repositoriesCoordinates: String)
 /**
  * An extension function that configures repositories and resolves artifacts denoted by the [Repository] and [DependsOn] annotations
  */
-suspend fun ExternalDependenciesResolver.resolveFromAnnotations(annotations: Iterable<Annotation>): ResultWithDiagnostics<List<File>> {
-    annotations.forEach { annotation ->
-        when (annotation) {
-            is Repository -> {
-                for (coordinates in annotation.repositoriesCoordinates) {
-                    if (!tryAddRepository(coordinates))
-                        return makeFailureResult("Unrecognized repository coordinates: $coordinates")
-                }
+suspend fun ExternalDependenciesResolver.resolveFromCollectedScriptAnnotations(
```

Comment:
See the comment about the class name above, this probably should be renamed too.

Reply:
✅